### ISSUE TITLE
GPU in 'Create New Notebook Server' wrong type

### ISSUE THREAD

**Issue by monoidus** [kubeflow/kubeflow#6114]
/kind bug

**What steps did you take and what happened:**
From the kubeflow-dashboard, when I create a new notebook server with the GPU field set to 1 (drop down selection), `Launch` shows me this error.

![gpuStringTypePopup](https://user-images.githubusercontent.com/6693081/131542620-dd94b811-ba0f-4a88-a399-9ec3939d2fb2.png)

**What did you expect to happen:**
I expected the notebook server to be launched without errors.

**Anything else you would like to add:**
 Seems to be similar to #5641 except for the GPU field.

**Environment:**

- Kubeflow version: dev_local (version number can be found at the bottom left corner of the Kubeflow dashboard):
  - [kubeflow-lite charm version 67](https://jaas.ai/u/kubeflow-charmers/kubeflow-lite)
- kfctl version: Not using this yet (use `kfctl version`):
- Kubernetes platform: [canonical kubernetes bundle #1202 via  juju](https://jaas.ai/canonical-kubernetes) on a local MAAS cluster.
- Kubernetes version: v1.21.4 (both client and server)
- OS: Ubuntu linux 20.04.3 LTS

**Comment by monoidus** [comment]
I keep getting this error. Ignored it and went ahead with only CPU field set. Turns out, using the `GPU=None` does not prevent the visibility of the GPU. The following pytorch code (using the pytorch-cuda-full image in notebook creation) detects my GPU.

```python
import torch
torch.cuda.is_available()
torch.cuda.current_device()
torch.cuda.device(0)
torch.cuda.device_count()
torch.cuda.get_device_name(0)
```

[this v0.6 setup link](https://v0-6.kubeflow.org/docs/notebooks/setup/) mentions that the `Extra Resources` field can be set to `{"nvidia.com/gpu": 2}` to **reserve** GPUs. Since this is for my home lab and I don't need to actually reserve any GPUs yet, this bug does not really block me.

However, bummer that the GPU field doesn't work and while there is a **Configurations** item in the UI, it does not allow freeform text but seems to be an empty drop-down.

**Comment by jbottum** [comment]
@monoidus did you get this resolved?  GPU configuration can be complex.      I am trying to understand your configuration i.e. are you using Kubeflow 0.6 with Kubernetes 1.21.4 with Canonical's charmed K8s and juju ?  Are you using any commercial vendors for support?

**Comment by monoidus** [comment]
@jbottum. Thanks for following up.

TL;DR: This particular problem I have not resolved. Seems to be a javascript issue in the UI similar to #5641. Looks scary when I see it but ends up not really blocking GPU usage so I can ignore it.

GPU config is actually all good. On a MAAS provisioned node, charmed-k8s installs the needed nvidia drivers and runtime (__ the containerd charm seems to doing it__). In my setup with 2 worker nodes (__one of which has the GPU__), `juju ssh kubernetes-worker/1 nvidia-smi` does report the card as expected. My KF install uses an overlay to place all the KF deployments  on the node with GPU (__32 core threadripper + 2080Ti__).

From what I understand, the UI's GPU field seems to be about reservation. Because of this bug, I leave the GPU field at None but the KF provided Jupyter servers correctly access the GPU (__I am using the `jupyter-pytorch-cuda-full` image__). `nvidia-smi` during a notebook run does show the underlying pytorch using almost 100% of the GPU. GPU access is good.

Sorry if the 0.6 link above confused the issue, the link was just for reference as it showed a way to perform GPU reservation without using the GUI. I am actually using charmed-kubeflow-lite (__Kubeflow 1.3 as far as I can tell__) on charmed-k8s (1.21.4): all with juju. No commercial support I'm afraid, this is for my personal home lab.

**Comment by ReggieCarey** [comment]
Warning: GPU selection chooses a node with GPU capability. If there is a problem as described here, it’s likely still present in a cluster with mode than one node type. 

Does this bug reflect problems with selecting MIG mixed mode GPUs?
